src/ctapipe/core/tests/test_tool.py

```diff
@@ -381,13 +381,42 @@ class ToolBad(Tool):
         def start(self):
             raise ValueError("1 does not equal 0.")
 
+    class CustomErrorNoExitCode(Exception):
+        pass
+
+    class CustomErrorWithExitCode(Exception):
+        exit_code = 42
+
+    class ToolCustomExceptionNoExitCode(Tool):
+        name = "CustomException"
+        description = "This tool raises a custom exception without an exit code."
+        custom_exceptions = (CustomErrorNoExitCode,)
```

Comment:
Why do we need to "register" these custom exceptions here? And if we do, why not use a mapping:

E.g.:
```
exception_code_mapping = {
    ValueError: 1,
    TypeError: 5,
}

```

Reply:
Exception registration is to distinguish between our own (means expected in certain circumstances) exceptions.
Considering the mapping, I believe exit codes shall be defined in the custom exceptions themselves, so writing them also here would be redundant.